test/integration/features/coolstore.feature

```diff
@@ -0,0 +1,137 @@
+     When executing "minishift start --memory 4096 --iso-url centos" succeeds
+     Then Minishift should have state "Running"
+
+  Scenario: User can create new namespace coolstore
+   Given Minishift has state "Running"
+    When executing "oc new-project coolstore" succeeds
+    Then stdout should contain
+     """
+     Now using project "coolstore"
+     """
+  
+  Scenario: User can create a template for coolstore 
+   Given Minishift has state "Running"
+    When executing "oc create -f $(template_dir)/coolstore.yaml" succeeds
+    Then stdout should contain
+     """
+     template "coolstore" created
+     """
+
+  Scenario: User can deploy the coolstore template to namespace coolstore 
+   Given Minishift has state "Running"
+    When executing "oc new-app --template=coolstore" succeeds
+    Then stdout should contain
+     """
+     Success
+     """
+
+  Scenario: Successfully deployed web-ui service to namespace coolstore 
+   Given Minishift has state "Running" 
+    When executing "oc rollout status deploymentconfig web-ui --watch" succeeds
+    Then stdout should contain
+     """
+     successfully rolled out
+     """
+
+  Scenario: Successfully deployed inventory service to namespace coolstore 
+   Given Minishift has state "Running" 
+    When executing "oc rollout status deploymentconfig inventory --watch" succeeds
+    Then stdout should contain
+     """
+     successfully rolled out
+     """
+
+  Scenario: Successfully deployed catalog service to namespace coolstore 
+   Given Minishift has state "Running" 
+    When executing "oc rollout status deploymentconfig catalog --watch" succeeds
+    Then stdout should contain
+     """
+     successfully rolled out
+     """
+
+  Scenario: Successfully deployed cart service to namespace coolstore 
+   Given Minishift has state "Running" 
+    When executing "oc rollout status deploymentconfig cart --watch" succeeds
+    Then stdout should contain
+     """
+     successfully rolled out
+     """
+
+  Scenario: Successfully rolled out gateway to namespace coolstore 
+   Given Minishift has state "Running" 
+    When executing "oc rollout status deploymentconfig coolstore-gw --watch" succeeds
+    Then stdout should contain
+     """
+     successfully rolled out
+     """
+
+  Scenario: Successfully deployed all services
+   Given Minishift has state "Running" 
+    When executing "oc status -v" succeeds
+    Then stdout should not contain
+     """
+     has failed
+     """
+
+  Scenario: Should be able to get the pod identifier for the API gateway 
+   Given Minishift has state "Running" 
+    When set variable "coolstore-gw" executing "oc get pods -o name -l application=coolstore-gw"
+    Then stdout should not be empty
+     And exitcode should equal "0"
+
+  Scenario: Able to interact with the catalog from the API gateway 
+   Given Minishift has state "Running"
+    When executing "oc rsh $(coolstore-gw) curl -sSL http://catalog:8080/api/products" succeeds
```

Comment:
internal services, like cart, catalog, etc are supposed to be only exposed by the api gateway and not publicly by a route.